Source/NSubstitute/Routing/RouteFactory.cs

```diff
@@ -57,6 +62,7 @@ public IRoute RecordCallSpecification(ISubstituteState state)
         {
             return new Route(new ICallHandler[] {
                 new RecordCallSpecificationHandler(state.PendingSpecification, state.CallSpecificationFactory, state.CallActions)
+                , new ClearLastTrackedCallHandler(state.LastCallTracker)
```

Comment:
The only reason why we track information about the last call is to later resolve specification for that "last call". During the "specifying call" we record call specification, which should be used for the subsequent `Returns()` and information about "last call" (which is now the call before the last) should not be used. 

Consider the following code sample:
```csharp
subs.MethodA(42); //register in LastCallTracker 
subs.MethodB(Arg.Any<int>()); //invoke `RecordCallSpecification` route.
```

As you can see, after the second invocation the `LastCallTracker` will hold `MethodA`, that is definitely obsoleted.
Even given that the `GetCallSpec` ignore last call information if pending specification is present ([see code](https://github.com/Zvirja/NSubstitute/blob/thread-local-last-call-info/Source/NSubstitute/Core/GetCallSpec.cs#L23)), I'd prefer to clear obsoleted data to avoid potential issues.